test/test_asymetric.py

```diff
@@ -151,6 +153,25 @@ def test_RSA_OAEP(self):
 
         self.assertEqual(text, plainText)
 
+    def test_RSA_OAEPwithAAD(self):
+        if self.SoftHSMversion < 2:
+            self.skipTest("RSA OAEP only supported by SoftHSM >= 2")
+
+        if self.manufacturer.startswith("SoftHSM"):
+            self.skipTest("SoftHSMv2 returns CKR_ARGUMENTS_BAD. 'AAD' not in expected format or unsupported.")
+
+        # RSA OAEP
+        plainText = "A test string"
+
+        aad = "sample aad".encode("utf-8")
+        mech = PyKCS11.RSAOAEPMechanism(PyKCS11.CKM_SHA_1, PyKCS11.CKG_MGF1_SHA1, aad)
+        cipherText = self.session.encrypt(self.pubKey, plainText, mech)
+        decrypted = self.session.decrypt(self.privKey, cipherText, mech)
+
+        text = "".join(map(chr, decrypted))
```

Comment:
Why not use `decrypted.decode('utf-8')`?

Reply:
Fixing. Thanks!